File: src/Entity/Hotel.php
<?php

namespace App\Entity;

use App\Repository\HotelRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;

/**
 * @ORM\Entity(repositoryClass=HotelRepository::class)
 * @ORM\Table(name="`hotel`")
 */

class Hotel
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Vérifier le nom! ")
     */
    private $nom;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Vérifier l'adresse! ")
     */
    private $adresse;

    /**
     * @ORM\Column(type="float")
     * @Assert\NotBlank(message="Vérifier le prix! ")

     */
    private $price;

    /**
     * @ORM\Column(type="string", length=255)
     * @Assert\NotBlank(message="Vérifier l'image! ")

     */
    private $image;

    /**
     * @ORM\Column(type="text")
     * @Assert\NotBlank(message="Vérifier la description! ")
     */
    private $description;

    /**
     * @ORM\Column(type="date")
     * @Assert\NotBlank(message="Vérifier la date! ")
     * @Assert\Expression("this.getDatedebut() < this.getDatefin()",message="Date début ne doit pas être dépassée la date fin")
     */
    private $datedebut;

    /**
     * @ORM\Column(type="date")
     * @Assert\Expression("this.getDateFin() > this.getDatedebut()",message="Date fin ne doit pas être antérieur à la date debut")
     */
    private $datefin;

    /**
     * @ORM\OneToMany(targetEntity=Reservation::class, mappedBy="hotel")
     */
    private $reservations;

    public function __construct()
    {
        $this->reservations = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getNom(): ?string
    {
        return $this->nom;
    }

    public function setNom(string $nom): self
    {
        $this->nom = $nom;

        return $this;
    }

    public function getAdresse(): ?string
    {
        return $this->adresse;
    }

    public function setAdresse(string $adresse): self
    {
        $this->adresse = $adresse;

        return $this;
    }

    public function getPrice(): ?float
    {
        return $this->price;
    }

    public function setPrice(float $price): self
    {
        $this->price = $price;

        return $this;
    }

    public function getImage()
    {
        return $this->image;
    }

    public function setImage($image): self
    {
        $this->image = $image;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(string $description): self
    {
        $this->description = $description;

        return $this;
    }

    public function getDatedebut(): ?\DateTimeInterface
    {
        return $this->datedebut;
    }

    public function setDatedebut(\DateTimeInterface $date): self
    {
        $this->datedebut = $date;

        return $this;
    }

    public function getDatefin(): ?\DateTimeInterface
    {
        return $this->datefin;
    }

    public function setDatefin(\DateTimeInterface $datefin): self
    {
        $this->datefin = $datefin;

        return $this;
    }

    /**
     * @return Collection|Reservation[]
     */
    public function getReservations(): Collection
    {
        return $this->reservations;
    }

    public function addReservation(Reservation $reservation): self
    {
        if (!$this->reservations->contains($reservation)) {
            $this->reservations[] = $reservation;
            $reservation->setHotel($this);
        }

        return $this;
    }

    public function removeReservation(Reservation $reservation): self
    {
        if ($this->reservations->removeElement($reservation)) {
            // set the owning side to null (unless already changed)
            if ($reservation->getHotel() === $this) {
                $reservation->setHotel(null);
            }
        }

        return $this;
    }
}

File: src/com/mycompany/myapp/entities/Hotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.mycompany.myapp.entities;

import java.util.Date;

/**
 *
 * @author user
 */
public class Hotel {
    private int id;
    private String nom,adresse,image,description;
    private double price;
    private Date Datedebut,Datefin;
    public Hotel(){
        
    }

    public Hotel(int id, String nom, String adresse, String image, String description,double price, Date Datedebut, Date Datefin) {
        this.id = id;
        this.nom = nom;
        this.adresse = adresse;
        this.image = image;
        this.description = description;
        this.price=price;
        this.Datedebut = Datedebut;
        this.Datefin = Datefin;
    }

    public Hotel(String nom, String adresse, double price,String description, Date Datedebut, Date Datefin) {
        this.nom = nom;
        this.adresse = adresse;
        this.description = description;
        this.price = price;
        this.Datedebut = Datedebut;
        this.Datefin = Datefin;
    }

    public Hotel(String nom, String adresse, String image, String description,double price, Date Datedebut, Date Datefin) {
        this.nom = nom;
        this.adresse = adresse;
        this.image = image;
        this.description = description;
        this.price=price;
        this.Datedebut = Datedebut;
        this.Datefin = Datefin;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    @Override
    public String toString() {
        return "Hotel{" + "id=" + id + ", nom=" + nom + ", adresse=" + adresse + ", image=" + image + ", description=" + description + ", price=" + price + ", Datedebut=" + Datedebut + ", Datefin=" + Datefin + '}';
    }

    
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getAdresse() {
        return adresse;
    }

    public void setAdresse(String adresse) {
        this.adresse = adresse;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public Date getDatedebut() {
        return Datedebut;
    }

    public void setDatedebut(Date Datedebut) {
        this.Datedebut = Datedebut;
    }

    public Date getDatefin() {
        return Datefin;
    }

    public void setDatefin(Date Datefin) {
        this.Datefin = Datefin;
    }

    
    
    
}
